CPAC/nuisance/utils/__init__.py

```diff
@@ -393,59 +393,73 @@ def generate_summarize_tissue_mask_ventricles_masking(nuisance_wf,
         mask_csf_with_lat_ven.inputs.expr = 'a*b'
         mask_csf_with_lat_ven.inputs.out_file = 'csf_lat_ven_mask.nii.gz'
 
-        ventricles_key = 'VentriclesToAnat'
-        if 'resolution' in regressor_descriptor:
-            ventricles_key += '_{}'.format(regressor_descriptor['resolution'])
-
-        if ventricles_key not in pipeline_resource_pool:
-
-            transforms = pipeline_resource_pool['Transformations']
-            
-            if use_ants is True:
+        # check if ventricle mask exists
+        ventricle_mask_check = pe.Node(util.Function(input_names = ['local_ventricle_mask'],
+                                                                function = check_ventricle_mask),
+                                                                name='check_ventricle')
+        nuisance_wf.connect(*(pipeline_resource_pool['Ventricles'] + (ventricle_mask_check, 'local_ventricle_mask'))) 
+        if ventricle_mask_check is True:  
```

Comment:
@HechengJin0 this will not work. You will not have the `check_ventricle_mask` function return in this step, but only on pipeline running (that, unfortunately, does not allow conditionals).

Reply:
Thank @anibalsolon 
Please see last commit.